### PR TITLE
openttd: music support

### PR DESCRIPTION
**Pull request by womfoo**
###### Motivation for this change
- fix music path for openmsx
- added wrapper script to play midi files based on https://wiki.archlinux.org/index.php/OpenTTD
- used pulseaudio as the default as I assume this is what most people use
###### Things done
- [x] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/openttd/default.nix [hunk 1]
```diff
@@ -18,6 +20,12 @@ let
     sha256 = "0qnmfzz0v8vxrrvxnm7szphrlrlvhkwn3y92b4iy0b4b6yam0yd4";
   };
 
+  playmidi = writeScriptBin "playmidi" ''
+    #!/bin/sh
+    trap "${procps}/bin/pkill fluidsynth" EXIT
+    ${fluidsynth}/bin/fluidsynth -a ${audioDriver} -i ${soundfont-fluid}/share/soundfonts/FluidR3_GM2-2.sf2 $*
```
Comment (abbradar) [inline pkgs/games/openttd/default.nix]: What will `fluidsynth` choose if no audio driver is specified? Is there some kind of "auto" option?

Comment (womfoo) [inline pkgs/games/openttd/default.nix]: Unfortunately there is no auto option and fluidsynth wont work without passing it a driver. I thought of checking for pulseaudio or alsa in the script but I dont have much experience in this.

Comment (womfoo) [inline pkgs/games/openttd/default.nix]: Also tried looking at timidity but I've read a lot of problems with it on pulseaudio

Comment (abbradar) [inline pkgs/games/openttd/default.nix]: Let's leave it at `alsa` then -- PulseAudio has a compatibility layer for ALSA apps but there's no way around.

Comment (womfoo) [inline pkgs/games/openttd/default.nix]: Done, rebased and tested OK for `alsa`

**Comment by womfoo** [comment]
Thanks for merging @abbradar !

**Comment by abbradar** [comment]
You are welcome ^_^

-- 
Nikolay.

**Pull request merged**